### ISSUE TITLE
No location field when editing proof

### ISSUE THREAD

**Issue by serpico** [openfoodfacts/open-prices-frontend#879]
### What
No location field when editing proof.

### Steps to reproduce the behavior:
Went to dashboard > My Proof > Filter > Hide proofs with prices  and found a proof without location ( maybe it's a rare case or the issue that generated the missing location has been solved, in that case I'm ok dismissing/closing the issue )
![Capture d’écran_2024-09-30_11-22-39](https://github.com/user-attachments/assets/080774a5-ca2c-45c7-be6f-a63e2e3fddf8)

Clicked to make the Edit card appear, Clicked on Edit in the menu.

I don't see a field to change, in that case, add the location.

![Capture d’écran_2024-09-30_11-21-33](https://github.com/user-attachments/assets/4f038b74-c18e-47ed-b0b5-597598539cb8)

### Expected behavior
Something allowing to change or add a location, preferably using OSM id

### Platform (Desktop, Mobile, Hunger Games)
 - OS: Linux
 - Platform : Laptop

**Comment by raphodn** [comment]
linked with #689

**Comment by raphodn** [comment]
In the first versions of the app (and in the mobile integration until quite recently), the proof did not have any location data (nor date & currency).
I can run a script to fill this data if there is at least 1 price attached to it.
But for proofs without prices, it's impossible automatically..

Indeed, if the proof has 0 prices, then everything should be editable...

**Comment by serpico** [comment]
I added a price to that particular proof, funnily it used the last location I used as a "default value". Feel free to ping me after running the script so I can try to edit...although I think we can't edit location yet even for "errorless" proof.

**Comment by raphodn** [comment]
Ah that's an issue..

It's this price right ? https://prices.openfoodfacts.org/prices/36477

I can fix it manually, then re-run the script

**Comment by serpico** [comment]
this one https://prices.openfoodfacts.org/app/products/3364699014104

**Comment by raphodn** [comment]
i had a look on OSM, there seems to be 2 Tang Frères in that neighborhood ^^

you've added prices for https://prices.openfoodfacts.org/app/locations/419 (https://www.openstreetmap.org/node/2621636797) but maybe it's the other one ? https://www.openstreetmap.org/node/252814154 :sweat_smile:

**Comment by serpico** [comment]
> i had a look on OSM, there seems to be 2 Tang Frères in that neighborhood ^^
> 
> you've added prices for https://prices.openfoodfacts.org/app/locations/419 (https://www.openstreetmap.org/node/2621636797) but maybe it's the other one ? https://www.openstreetmap.org/node/252814154 😅

You're the man !!! good catch !!!
Yeah I remember it was the early days and I struggled with Komoot's results...😅

it's definitely 252814154, if you can transfer all there, I clean up the light bulb which is another mistake :)

**Comment by serpico** [comment]
https://prices.openfoodfacts.org/app/locations/420 all these are in fact 252814154 ( OSM id ) as well, don't know I you can do something, or we can wait till we're a little further along in the possibly for users to amend their contributions, it's a good use case for testing those features

**Comment by raphodn** [comment]
ok no worries :)
I have a small utility script to help move a proof's location (and its attached prices), i just did it for both 419 & 420.
They are now here : https://prices.openfoodfacts.org/app/locations/1113 (except your lightbulb, the proof seems different from the price)

**Comment by serpico** [comment]
yeah for 3601029959705 , the light bulb, both the date and location are wrong ( it was me ), what do you think we keep it as is, until registered users are allowed/permitted to change the date and location of a price ?

**Comment by raphodn** [comment]
> until registered users are allowed/permitted to change the date and location of a price

I'd say that option would only be available at the proof level. and it would then impact the X prices linked to it.
And yeah in #839 we need to find ways for users like you with a LOT of proofs :innocent:

**Comment by raphodn** [comment]
Bumping this up, as the backend allows this: https://github.com/openfoodfacts/open-prices-frontend/issues/879
